### PR TITLE
Public patroller

### PR DESCRIPTION
**Pull request by akinsey**
references #709

1) There should now be an setting on the settings page to view the patrol view as any user.
2) Depending on your role you should see different actions:
  a) If you are a mod you can moderate posts by newbies that happen to be in a board you moderate. 
  b) If you are a global mod or admin you should be able to moderate all posts on the patroller page
  c) If you are a patroller you should be able to access limited moderation controls
  d) If you are a standard user you should be able to report the post (Once we merge the new editor code we will make it so you can reply/quote from this view directly)

### CODE REVIEW

**Comment by Rizzrack09** [comment]
sounds awesome :)

File: modules/bct-patroller/db/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var path = require('path');
+
+module.exports = {
+  patrol: require(path.normalize(__dirname + '/patrol')),
```
Comment (unenglishable) [inline modules/bct-patroller/db/index.js]: extra comma

File: modules/bct-patroller/routes/patrol.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      query: Joi.object({
+        page: Joi.number().integer().min(1),
+        limit: Joi.number().integer().min(1).max(200).default(25)
+      }).without('start', 'page')
+    },
+    pre: [
+      { method: (request) => request.server.methods.hooks.preProcessing(request) },
+      [
+        { method: (request) => request.server.methods.hooks.parallelProcessing(request), assign: 'parallelProcessed' },
+        { method: processing, assign: 'processed' },
```
Comment (unenglishable) [inline modules/bct-patroller/routes/patrol.js]: extra comma

File: modules/ept-breadcrumbs/db/getBreadcrumbs.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    category: request.db.categories.find,
+    thread: request.db.threads.breadcrumb,
+    post: request.db.posts.find
+  };
+
+  // Type enum
+  var type = {
+    board: 'board',
+    category: 'category',
+    thread: 'thread',
```
Comment (unenglishable) [inline modules/ept-breadcrumbs/db/getBreadcrumbs.js]: extra comma

File: modules/ept-breadcrumbs/db/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+var path = require('path');
+
+module.exports = {
+  getBreadcrumbs: require(path.normalize(__dirname + '/getBreadcrumbs')),
```
Comment (unenglishable) [inline modules/ept-breadcrumbs/db/index.js]: extra comma

**Pull request merged**